### PR TITLE
Update: Add linebreak style option to eol-last (fixes #4148)

### LINKED ISSUE #4148: Bug: eol-last autofix doesn't take linebreak-style into account
eol-last autofix always inserts '\n' at the end of the file regardless of the file's linebreak-style

e.g. 
input
```js
var a = 123;\r\nvar b = 456;
```

expected
```js
var a = 123;\r\nvar b = 456\r\n;
```

result
```js
var a = 123;\r\nvar b = 456\n;
```

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/27442380-bug-eol-last-autofix-doesn-t-take-linebreak-style-into-account?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by alberto**

### CODE REVIEW

**Comment by ilyavolodin** [comment]
LGTM

**Comment by alberto** [comment]
Updated title to reflect what the PR does now.

File: docs/rules/eol-last.md [hunk 1]
```diff
@@ -34,3 +34,7 @@ function doSmth() {
 }
 // spaces here
 ```
+
+### Options
```
Comment (nzakas) [inline docs/rules/eol-last.md]: Can you show an example with the rule option?

Comment (alberto) [inline docs/rules/eol-last.md]: What kind of example would you expect? The option only takes effect while fixing, so it is not apparent in the example.

**Comment by gyandeeps** [comment]
Looks like an "Update:" rather than a plain "Fix:" because we are introducing an option here.
Thoughts?

**Comment by nzakas** [comment]
@gyandeeps agree. This is an "Update"

**Comment by alberto** [comment]
Commit message updated.

File: docs/rules/eol-last.md (hunk 1 above)
Comment (nzakas) [inline docs/rules/eol-last.md]: Show the actual rule configuration so people can copy-paste it.

**Comment by adrienverge** [comment]
Could you change the module exports schema so we can add other options in the future? (I plan to add a "max" option to specify the max number of newlines at the end of the file.)

Something like:
```
"eol-last": [2, {"type": "unix"}]
```
would be great.

**Pull request merged**